src/box/internal.tsx

```diff
@@ -24,7 +25,12 @@ export default function InternalBox({
   __internalRootRef = null,
   ...props
 }: InternalBoxProps) {
-  const baseProps = getBaseProps(props);
+  const wrapperHeadingId = useContext(ScrollbarLabelContext);
```

Comment:
Box is one of our most used components in the system. I'm a little scared of adding a React context to it. Have you checked this for potential performance degredation?

Reply:
Box is definitely tricky, and there are more concerns (like pointing aria to a non-existing id). So I changed the implementation to provide an id from the header component.
But it might still worth checking for performance issues in a dev pipeline first